Library/OcAppleKeysLib/OcAppleKeysLib.c

```diff
@@ -12,39 +12,54 @@
   WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 **/
 
+#include <Base.h>
+
 #include <Library/OcAppleKeysLib.h>
 
-CONST APPLE_PK_ENTRY PkDataBase[NUM_OF_PK] = {
+#pragma pack(push, 1)
+
+typedef struct {
+  ///
+  /// RSA Public Key header structure.
+  ///
+  RSA_PUBLIC_KEY_HDR Hdr;
+  ///
+  /// The Modulus and Montgomery's R^2 mod N in little endian byte order.
+  ///
+  union {
+    UINT8            Bytes[512];
+    UINT64           QWords[32];
+  }                  Data;
+} RSA_PUBLIC_KEY_2048;
+
+#pragma pack(pop)
+
+OC_STATIC_ASSERT (
+  sizeof (RSA_PUBLIC_KEY_2048) == sizeof (RSA_PUBLIC_KEY_HDR) + 512,
```

Comment:
I really do not like that 512. Same with 32. 32 * 8 = 256 bytes, 512 * 8 = 4096 bits. To me it is not obvious, how that works, please leave a comment and make some named constants. Quite possible there is a typo here.

Reply:
It's because it stores N and RSqrMod, which both have equal size.

Reply:
Well, the claim still holds, as this is union and I still do not see a clarification in the code why it is fine to have 256 and 512 bytes in a union as is.

Reply:
Improved at https://github.com/acidanthera/OcSupportPkg/commit/33b60c588a3880323589754fe18084a4c31bfc90

Reply:
Really fixed in https://github.com/acidanthera/OcSupportPkg/commit/8cba450bfd99bdf659e43c1cb5ba4f07f0a45c55 I guess.